lib/std/fmt.zig

```diff
@@ -488,25 +488,14 @@ pub fn formatType(
         },
         .Enum => |enumInfo| {
             try writer.writeAll(@typeName(T));
-            if (enumInfo.is_exhaustive) {
+            if (enumInfo.is_exhaustive or std.enums.hasName(value)) {
```

Comment:
Note, even though `hasName` will return true if `enumInfo.is_exhaustive` is true, this messes up this function because zig no longer evaluates that codepath at comptime, which causes it to evaluate the `else` branch which does not compile in some cases.